### PR TITLE
Add demucs with transformer 2

### PR DESCRIPTION
**Pull request by mosem**

### CODE REVIEW

**Review by Or-Tal — CHANGES_REQUESTED** [review]
could you split to several pull requests? I can't follow this in a reasonable time

File: conf/config.yaml [hunk 1]
```diff
@@ -66,8 +66,9 @@ world_size:
 # Hydra config
```
Comment (Or-Tal) [inline conf/config.yaml:66]: this should match master's main_config

File: conf/experiment/demucs_skipless_1.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+experiment_name: demucs_skipless_1
```
Comment (Or-Tal) [inline conf/experiment/demucs_skipless_1.yaml:1]: lacking #@package

File: conf/experiment/demucs_skipless_adversarial_1.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+experiment_name: demucs_skipless_adversarial_1
```
Comment (Or-Tal) [inline conf/experiment/demucs_skipless_adversarial_1.yaml:1]: lacking #@package

File: conf/experiment/demucs_with_transformer_1.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+experiment_name: demucs_with_transformer_1
```
Comment (Or-Tal) [inline conf/experiment/demucs_with_transformer_1.yaml:1]: lacking #@package

File: conf/experiment/demucs_with_transformer_adversarial_1.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+experiment_name: demucs_with_transformer_adversarial_1
```
Comment (Or-Tal) [inline conf/experiment/demucs_with_transformer_adversarial_1.yaml:1]: lacking #@Package

File: denoiser/models/autoencoder_composer.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        self.decoder = decoder
+        self.skips = skips
+
+    def estimate_valid_length(self, input_length):
+        encoder_output_length = self.encoder.estimate_output_length(input_length)
+        attention_output_length = self.attention_module.estimate_output_length(encoder_output_length)
+        decoder_output_length = self.decoder.estimate_output_length(attention_output_length)
+        return decoder_output_length
+
+    def forward(self, signal):
```
Comment (Or-Tal) [inline denoiser/models/autoencoder_composer.py:21]: to my opinion, the whole normalization should be dealt with in this function, same for re-multiplying the output by std.

File: test_denoiser.py [hunk 7]
```diff
@@ -12,28 +12,37 @@
...
-               'experiment.segment=2', 'experiment.stride=2', 'ddp=0',
-               'experiment.batch_size=1',
-               'eval_every=1', 'epochs=1', f'hydra.run.dir={OUTPUT_DIR}', f'device={device}'],
+                'experiment.segment=2', 'experiment.stride=2', 'ddp=0',
+                'experiment.batch_size=1',
+                'eval_every=1', 'epochs=1', f'hydra.run.dir={OUTPUT_DIR}', f'device={device}'],
     'demucs_hifi': ['train.py', 'dset=valentini_dummy', 'experiment=demucs_hifi',
-               'experiment.segment=2', 'experiment.stride=2','ddp=0', 'experiment.scale_factor=2',
-               'eval_every=1', 'epochs=1', f'hydra.run.dir={OUTPUT_DIR}', f'device={device}'],
+                'experiment.segment=2', 'experiment.stride=2','ddp=0', 'experiment.scale_factor=2',
```
Comment (Or-Tal) [inline test_denoiser.py:35]: demucs hifi should not be here.. it's my bad but could you please delete it?

**Review by Or-Tal — CHANGES_REQUESTED** [review]
Hi, Please see comments

File: conf/config.yaml (hunk 1 above)
Comment (mosem) [reply conf/config.yaml]: merged with master. done.

File: denoiser/models/autoencoder_composer.py (hunk 6 above)
Comment (mosem) [reply denoiser/models/autoencoder_composer.py:23]: done.

File: test_denoiser.py (hunk 7 above)
Comment (mosem) [reply test_denoiser.py]: done. shouldn't we remove other code as well? such as in batch_solver_factory.py, hifi_gan_loss_functions,py,  demucs_hifi_generator.py, and demucs_hifi_bs.py?

Comment (Or-Tal) [reply test_denoiser.py]: will do in a separate pull request

**Pull request merged**